### PR TITLE
Create submodule `MathProg`

### PR DESCRIPTION
**Pull request by guimarqu**

### CODE REVIEW

File: src/Coluna.jl [hunk 1]
```diff
@@ -1,56 +1,47 @@
...
 
+# submodules
+export Containers, MathProg
+
 # Base functions for which we define more methods in Coluna
 import Base: isempty, hash, isequal, length, iterate, getindex, lastindex,
     getkey, delete!, setindex!, haskey, copy, promote_rule, convert, isinteger,
-    push!, filter
+    push!, filter, diff
+
```
Comment (vitornesello) [inline src/Coluna.jl]: Two blank lines here?

File: src/Coluna.jl [hunk 2]
```diff
@@ -1,56 +1,47 @@
...
-include("formulation.jl")
-include("optimizerwrappers.jl")
-include("clone.jl")
-include("reformulation.jl")
-include("projection.jl")
-include("problem.jl")
 include("node.jl")
-include("decomposition.jl")
-include("MOIinterface.jl")
+
```
Comment (vitornesello) [inline src/Coluna.jl]: Two lines here

File: src/MathProg/MathProg.jl [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+       unsafe_getbestprimalsol, getcurub, setcurrhs!, getcurcost, getcost,
+       _active_BendSpSlackFirstStage_var_, getconstrcounter, setprimaldualbendspsol!,
+       defaultprimalboundvalue, set_lp_primal_bound!, getpereneub, _active_,
+       getperenecost, update_ip_primal_bound!, getprimaldwspsolmatrix, _active_explicit_,
+       _rep_of_orig_var_, getcursense, getcurkind, find_owner_formulation,
+       setfeasibilitystatus!, setterminationstatus!, get_dw_pricing_sps, 
+       setprimalsol!, setdualsol!, getsortuid, setcol_from_sp_primalsol!,
+       get_benders_sep_sps, setcut_from_sp_dualsol!, getprimalsolmatrix
+
+
```
Comment (vitornesello) [inline src/MathProg/MathProg.jl]: I guess more more than one blank line is acceptable in such cases?

File: src/MathProg/reformulation.jl [hunk 4]
```diff
@@ -4,7 +4,7 @@
 Representation of a formulation which is solved by Coluna using a decomposition approach. All the sub-structures are defined within the struct `Reformulation`.
 """
 mutable struct Reformulation <: AbstractFormulation
-    strategy::GlobalStrategy
+    strategy::AbstractGlobalStrategy # to do : remove the link with algorithms
```
Comment (vitornesello) [inline src/MathProg/reformulation.jl]: I cannot understand this todo comment

Comment (guimarqu) [reply src/MathProg/reformulation.jl]: The strategy attribute will be removed

**Pull request merged**